core/src/main/java/org/elasticsearch/plugins/DiscoveryPlugin.java

```diff
@@ -52,4 +56,6 @@
     default NetworkService.CustomNameResolver getCustomNameResolver(Settings settings) {
         return null;
     }
+
+    default Set<Class<? extends ZenPing>> getZenPings(Settings settings) { return Collections.emptySet(); }
```

Comment:
Sure, but then instead of having a method in the DiscoveryPlugin, we have a protected method, dedicated to zen pings. We don't do that for anything else so I think it's a shame to introduce a new pattern for something that's going away anyway when guice is removed. At which point, if I understand your plan correctly, the test plugin will supply a complete implementation with the zen pings replaced.